src/cross_section/src/cross_section.cpp

```diff
@@ -0,0 +1,318 @@
+  else if (crossSections.size() == 1)
+    return crossSections[0];
+
+  auto subjs = crossSections[0].paths_;
+  int n_clips = 0;
+  for (int i = 1; i < crossSections.size(); ++i) {
+    n_clips += crossSections[i].paths_.size();
+  }
+  auto clips = C2::PathsD();
+  clips.reserve(n_clips);
+  for (int i = 1; i < crossSections.size(); ++i) {
+    auto ps = crossSections[i].paths_;
+    clips.insert(clips.end(), ps.begin(), ps.end());
+  }
+
+  auto ct = cliptype_of_op(op);
+  auto res =
+      C2::BooleanOp(ct, C2::FillRule::Positive, subjs, clips, precision_);
+  return CrossSection(res);
+}
+
+/**
+ * Shorthand for Boolean Union.
+ */
+CrossSection CrossSection::operator+(const CrossSection& Q) const {
+  return Boolean(Q, OpType::Add);
+}
+
+/**
+ * Shorthand for Boolean Union assignment.
+ */
+CrossSection& CrossSection::operator+=(const CrossSection& Q) {
+  *this = *this + Q;
+  return *this;
+}
+
+/**
+ * Shorthand for Boolean Difference.
+ */
+CrossSection CrossSection::operator-(const CrossSection& Q) const {
+  return Boolean(Q, OpType::Subtract);
+}
+
+/**
+ * Shorthand for Boolean Difference assignment.
+ */
+CrossSection& CrossSection::operator-=(const CrossSection& Q) {
+  *this = *this - Q;
+  return *this;
+}
+
+/**
+ * Shorthand for Boolean Intersection.
+ */
+CrossSection CrossSection::operator^(const CrossSection& Q) const {
+  return Boolean(Q, OpType::Intersect);
+}
+
+/**
+ * Shorthand for Boolean Intersection assignment.
+ */
+CrossSection& CrossSection::operator^=(const CrossSection& Q) {
+  *this = *this ^ Q;
+  return *this;
+}
+
+CrossSection CrossSection::Translate(glm::vec2 v) {
+  auto ps = C2::TranslatePaths(paths_, v.x, v.y);
+  return CrossSection(ps);
+}
+
+CrossSection CrossSection::Rotate(float degrees) {
+  auto rotated = C2::PathsD();
+  rotated.reserve(paths_.size());
+  auto s = sind(degrees);
+  auto c = cosd(degrees);
+  for (auto path : paths_) {
+    auto r = C2::PathD();
+    r.reserve(path.size());
+    for (auto p : path) {
+      auto rx = (p.x * c) - (p.y * s);
+      auto ry = (p.y * c) + (p.x * s);
+      r.push_back(C2::PointD(rx, ry));
+    }
+    rotated.push_back(r);
+  }
+  return CrossSection(rotated);
+}
+
+CrossSection CrossSection::Scale(glm::vec2 scale) {
+  auto scaled = C2::PathsD();
+  scaled.reserve(paths_.size());
+  for (auto path : paths_) {
+    auto s = C2::PathD();
+    s.reserve(path.size());
+    for (auto p : path) {
+      s.push_back(C2::PointD(p.x * scale.x, p.y * scale.y));
+    }
+    scaled.push_back(s);
+  }
+  return CrossSection(scaled);
+}
+
+CrossSection CrossSection::Mirror(glm::vec2 ax, bool reverse) {
+  auto mirrored = C2::PathsD();
+  mirrored.reserve(paths_.size());
+  for (auto path : paths_) {
+    auto sz = path.size();
+    auto m = C2::PathD(sz);
+    for (int i = 0; i < sz; ++i) {
+      auto v = v2_of_pd(path[i]);
+      int idx = reverse ? sz - 1 - i : i;
+      m[idx] = v2_to_pd(ax * (2 * glm::dot(v, ax) / glm::dot(ax, ax)) - v);
```

Comment:
Perhaps we should have a check to make sure that `ax` length is non-zero and throw an exception/different return value when it is? So people writing bindings or using it as the backend don't have to write too much code for checking.

Reply:
Would returning the same CrossSection back be appropriate (no axis to mirror over)? Or an empty CrossSection? Speaking of, should probably add a method to check if it's empty.

Reply:
I've changed it to return an empty CrossSection for the moment (coinciding with the addition of IsEmpty). Please let me know if you guys think that it should be the unchanged input instead.